File: test/src/main/java/com/shipping/service/FunctionService.java
package com.shipping.service;

import com.shipping.domain.sys.Function;
import com.shipping.repository.sys.FunctionRepository;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.sql.ResultSet;
import java.sql.Statement;
import java.util.Objects;

/**
 * <p>Created by xuweinan on 2017/9/4.</p>
 *
 * @author xuweinan
 */
@Service
public class FunctionService {
    private static final Logger logger = LoggerFactory.getLogger(FunctionService.class);

    @Autowired
    private FunctionRepository functionDao;

    public void add(Function func) {
        if (Objects.nonNull(func.getParent())) {
            Function parent = functionDao.get(func.getParent().getId());
            functionDao.save(func, true);
            functionDao.doWork(connection -> {Statement stmt = connection.createStatement();
            ResultSet rs = stmt.executeQuery("SELECT * FROM sys_function");});
        }
    }
}

File: test/src/main/kotlin/com/shipping/service/FunctionService.kt
package com.shipping.service

import com.shipping.domain.enums.OrganizationTypeE
import com.shipping.domain.sys.Function
import mu.KLogging
import org.spin.data.extend.RepositoryContext
import org.spin.data.rs.MapRowMapper
import org.spin.web.annotation.RestfulMethod
import org.spin.web.annotation.RestfulService
import org.springframework.beans.factory.annotation.Autowired
import javax.transaction.Transactional

/**
 *
 * Created by xuweinan on 2017/9/4.
 *
 * @author xuweinan
 */
@RestfulService
class FunctionService {
    companion object : KLogging()

    @Autowired
    private lateinit var repoCtx: RepositoryContext

    @RestfulMethod
    @Transactional
    fun add(func: Function) {
        func.parent?.let {
            OrganizationTypeE.GROUP
            repoCtx.get(Function::class.java, func.parent!!.id)
            repoCtx.save(func, true)
            val list: List<Map<String, Any?>> = repoCtx.doReturningWork("SELECT * FROM sys_function") {
                MapRowMapper().extractData(it)
            }
            list
        }
    }
}
